### PR TITLE
upgrade to Llama2

### LINKED ISSUE #114: upgrade llama support to llama2

### PR DESCRIPTION
**Pull request by ArshErgon**
This PR close #114 I have upgraded the support for Llama2

### CODE REVIEW

**Comment by cartazio** [comment]
You need to update the default model url! And the registration name in the @ annotation

**Comment by cartazio** [comment]
Also bump the min version number for the transformer library to reflect that class.  Once that’s sorted looks good to me

**Comment by ArshErgon** [comment]
hey @cartazio I've made the requested changes, I have change the @ annotation name and also bump the version of transformer, but I dont understand the default mode url, do you mean `model_uri`?

**Comment by cartazio** [comment]
The default model url needs to be updated to point to a llama2.  Once that’s done it’s good to go

File: src/llm_vm/client.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
     default_big_model= conf.settings.big_model
 
 if conf.settings.small_model is not  None:
-    default_small_model= "pythia"
+    default_small_model= "llama2"
```
Comment (cartazio) [inline src/llm_vm/client.py:16]: this is wrong

File: src/llm_vm/onsite_llm.py [hunk 2]
```diff
@@ -299,9 +299,9 @@ class Small_Local_LLama(Base_Onsite_LLM):
     model_uri="openlm-research/open_llama_3b_v2"
```
Comment (cartazio) [inline src/llm_vm/onsite_llm.py:299]: this is the URL you need to fix

File: src/llm_vm/client.py (hunk 1 above)
Comment (cartazio) [reply src/llm_vm/client.py:16]: undo this :)

File: src/llm_vm/onsite_llm.py (hunk 2 above)
Comment (ArshErgon) [reply src/llm_vm/onsite_llm.py:299]: Updated the requested URL to Llama-2-70b-hf
Source:https://huggingface.co/blog/llama2 (https://huggingface.co/meta-llama/Llama-2-70b-hf)

Comment (cartazio) [reply src/llm_vm/onsite_llm.py]: What I mean is: defaults for models should always be the smallest version possible so anyone can run the code locally

**Comment by cartazio** [comment]
After the mode selection tweak this should be good for merging

File: src/llm_vm/onsite_llm.py [hunk 3]
```diff
@@ -296,12 +296,12 @@ class Small_Local_LLama(Base_Onsite_LLM):
         tokenizer_loader: Loads the tokenizer into memory
         generate: Generates a response from a given prompt with the loaded LLM and tokenizer
     """
-    model_uri="openlm-research/open_llama_3b_v2"
+    model_uri="meta-llama/Llama-2-70b-hf"
```
Comment (cartazio) [inline src/llm_vm/onsite_llm.py:299]: changes this to 7B or whatever its called and i'll merge it in

Comment (ArshErgon) [reply src/llm_vm/onsite_llm.py:299]: I have downgrade the version to the lowest, as here mentioned: https://huggingface.co/meta-llama

Comment (cartazio) [reply src/llm_vm/onsite_llm.py]: 70b is much bigger than 7b

Comment (cartazio) [reply src/llm_vm/onsite_llm.py]: woops nvm

File: src/llm_vm/onsite_llm.py [hunk 4]
```diff
@@ -296,12 +296,12 @@ class Small_Local_LLama(Base_Onsite_LLM):
         tokenizer_loader: Loads the tokenizer into memory
         generate: Generates a response from a given prompt with the loaded LLM and tokenizer
     """
-    model_uri="openlm-research/open_llama_3b_v2"
+    model_uri="meta-llama/Llama-2-7b"
```
Comment (cartazio) [inline src/llm_vm/onsite_llm.py:299]: perfect

**Pull request merged**